kfs-prje/src/main/java/edu/arizona/kfs/module/prje/service/impl/PRJEServiceImpl.java

```diff
@@ -0,0 +1,857 @@
+            creditEntry.setUniversityFiscalYear(fiscalYear);
+
+            creditEntry.setTransactionDate(transactionDate);
+            creditEntry.setTransactionLedgerEntryAmount(amount.abs());
+            creditEntry.setTransactionLedgerEntrySequenceNumber(lastOriginSequence++);
+            creditEntry.setTransactionScrubberOffsetGenerationIndicator(true);
+
+            String description = generateCreditDescription(transferRecord);
+            creditEntry.setTransactionLedgerEntryDescription(description);
+            
+            return creditEntry;
+        }
+        
+        private void writeStagingEntries(List<OriginEntryFull> originEntries) {
+            LOG.info("Writing Origin Entries to Staging");
+            
+            String dataFilename = getBatchFileDirectoryName() + File.separator + DATA_FILE;
+            String doneFilename = getBatchFileDirectoryName() + File.separator + DONE_FILE;
+            
+            try {
+                PrintStream ps = new PrintStream(dataFilename);
+            
+                for ( OriginEntryFull originEntry : originEntries ) {
+                    
+                    // Do a little bit of house cleaning of entries
+                    if ( StringUtils.isEmpty(originEntry.getSubAccountNumber()) ) {
+                        originEntry.setSubAccountNumber(KFSConstants.getDashSubAccountNumber());
+                    }
+                    if ( StringUtils.isEmpty(originEntry.getProjectCode()) ) {
+                        originEntry.setProjectCode(KFSConstants.getDashProjectCode());
+                    }
+                    if ( StringUtils.isEmpty(originEntry.getFinancialSubObjectCode()) ) {
+                        originEntry.setFinancialSubObjectCode(KFSConstants.getDashFinancialSubObjectCode());
+                    }
+                    
+                    ps.printf("%s\n", originEntry.getLine());
+                    
+                    LOG.debug("- FLAT FILE: "+originEntry.getLine());
+                }
+                
+                ps.close();
+                
+                // Touch the DONE file
+                File doneFile = new File(doneFilename);
+                if ( !doneFile.exists() ) {
+                    doneFile.createNewFile();
+                }
+            }
+            catch ( IOException e ) {
+                throw new RuntimeException(e);
+            }
+        }
+        
+        private KualiDecimal calculateBaseAccountAmount(PRJEBaseAccount baseAccount) {
+            KualiDecimal curBalance = ZERO;
+            LOG.debug("-- Processing Base Account "+baseAccount.getBaseAccount());
+            
+            // Retrieve the balances for this baseAccount
+            BusinessObjectService bos = getBusinessObjectService();
+            Map<String, String> args = new HashMap<String, String>();
+            args.put("universityFiscalYear", Integer.toString(fiscalYear));
```

Comment:
Should use KFSPropertyConstants.UNIVERSITY_FISCAL_YEAR, KFSPropertyConstants.CHART_OF_ACCOUNTS_CODE and KFSPropertyConstants.ACCOUNT_NUMBER here.